src/components/MenuPanel/MenuPanelAppsLoader.js

```diff
@@ -0,0 +1,142 @@
+import React from 'react'
+import styled from 'styled-components'
+import { spring, Motion } from 'react-motion'
+import { spring as springConf, theme } from '@aragon/ui'
+import { lerp } from '../../math-utils'
+
+const SPRING = springConf('fast')
+
+class MenuPanelAppsLoader extends React.Component {
+  state = {
+    hideLoader: false,
+  }
+  componentWillReceiveProps({ loading }) {
+    if (loading && !this.props.loading) {
+      this.setState({ hideLoader: false })
+    }
+  }
+  handleRest = () => {
+    if (!this.props.loading) {
+      // To trigger the Motion transition
+      setTimeout(() => {
+        this.setState({ hideLoader: true })
+      }, 0)
+    }
+  }
+  render() {
+    const { children, loading } = this.props
+    const { hideLoader } = this.state
+    return (
+      <Motion
+        onRest={this.handleRest}
+        defaultStyle={{ prepareProgress: 0, hideProgress: 0 }}
+        style={{
+          prepareProgress: spring(Number(!loading), SPRING),
+          hideProgress: spring(Number(hideLoader), SPRING),
+        }}
+      >
+        {({ prepareProgress, hideProgress }) => (
+          <Main>
+            <Loader>
+              <LoaderBackground style={{ opacity: prepareProgress }} />
+              <LoaderContent>
+                <LoadCircle spin={loading}>
+                  <LoadCircleIn spin={loading} />
+                </LoadCircle>
+                {loading ? 'Loading apps…' : 'Apps loaded.'}
+              </LoaderContent>
+            </Loader>
+            <Apps
+              style={{
+                height: `${lerp(prepareProgress, 40, children.length * 40)}px`,
+                transform: `translateX(-${(1 - hideProgress) * 100}%)`,
+              }}
+            >
+              {children}
+            </Apps>
+          </Main>
+        )}
+      </Motion>
+    )
+  }
+}
+
+const LoadCircle = styled.div`
+  position: relative;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 22px;
+  height: 22px;
+  margin-right: 15px;
+  @keyframes spin {
+    from {
+      transform: rotate(0deg);
+    }
+    to {
+      transform: rotate(360deg);
+    }
+  }
+  animation: ${({ spin }) => (spin ? 'spin 1s infinite linear' : 'none')};
+`
+
+const LoadCircleIn = styled.div`
+  position: relative;
+  overflow: ${({ spin }) => (spin ? 'hidden' : 'visible')};
+  width: 11px;
```

Comment:
I imagine this was deliberate, but wanted to note that `11px` means that one of the "holes" is bigger than the other (`10px` or `12px` would have them equal):

![image](https://user-images.githubusercontent.com/4166642/38017791-5eb9e18a-3273-11e8-8bce-088318453400.png)

I think it looks awesome the way it is, but just in case this was something that slipped by :).

Reply:
Ah yes, Chrome (still 😩) doesn’t know how to render borders on subpixels… changing it to `10px`, thanks!